File: source/se/krka/kahlua/integration/LuaSuccess.java
// Decompiled on Sat Jan 17 08:24:00 EST 2026 with Zomboid Decompiler v0.2.3 using Vineflower.
package se.krka.kahlua.integration;

public class LuaSuccess extends LuaReturn {
    LuaSuccess(Object[] objects) {
        super(objects);
    }

    @Override
    public boolean isSuccess() {
        return true;
    }

    @Override
    public Object getErrorObject() {
        throw new UnsupportedOperationException("Not valid when isSuccess is true");
    }

    @Override
    public String getErrorString() {
        throw new UnsupportedOperationException("Not valid when isSuccess is true");
    }

    @Override
    public String getLuaStackTrace() {
        throw new UnsupportedOperationException("Not valid when isSuccess is true");
    }

    @Override
    public RuntimeException getJavaException() {
        throw new UnsupportedOperationException("Not valid when isSuccess is true");
    }
}

File: PZ-Switch/src/se/krka/kahlua/integration/LuaSuccess.cpp
#include "se/krka/kahlua/integration/LuaSuccess.h"

namespace se {
namespace krka {
namespace kahlua {
namespace integration {

bool LuaSuccess::isSuccess() {
 // TODO: Implement isSuccess
 return false;
}

void *LuaSuccess::getErrorObject() {
 // TODO: Implement getErrorObject
 return nullptr;
}

std::string LuaSuccess::getErrorString() {
 // TODO: Implement getErrorString
 return "";
}

std::string LuaSuccess::getLuaStackTrace() {
 // TODO: Implement getLuaStackTrace
 return "";
}

std::runtime_error LuaSuccess::getJavaException() {
 // TODO: Implement getJavaException
 return nullptr;
}

} // namespace integration
} // namespace kahlua
} // namespace krka
} // namespace se
